File: src/java/org/elastos/hive/vault/database/UpdateResult.java
package org.elastos.hive.vault.database;

import com.google.gson.annotations.SerializedName;

public class UpdateResult {
	private Boolean acknowledged;
	@SerializedName("matched_count")
	private Integer matchedCount;
	@SerializedName("modified_count")
	private Integer modifiedCount;
	@SerializedName("upserted_id")
	private String upsertedId;

	public void setAcknowledged(Boolean acknowledged) {
		this.acknowledged = acknowledged;
	}

	public void setMatchedCount(Integer matchedCount) {
		this.matchedCount = matchedCount;
	}

	public void setModifiedCount(Integer modifiedCount) {
		this.modifiedCount = modifiedCount;
	}

	public void setUpsertedId(String upsertedId) {
		this.upsertedId = upsertedId;
	}

	public Boolean getAcknowledged() {
		return acknowledged;
	}

	public Integer getMatchedCount() {
		return matchedCount;
	}

	public Integer getModifiedCount() {
		return modifiedCount;
	}

	public String getUpsertedId() {
		return upsertedId;
	}
}

File: src/service/database/updateresult.ts
export class UpdateResult {
	private acknowledged: boolean;
	private matched_count: number;
	private modified_count: number;
	private upserted_id: string;

	setAcknowledged(acknowledged: boolean) {
		this.acknowledged = acknowledged;
		return this;
	}

	setMatchedCount(matchedCount: number) {
		this.matched_count = matchedCount;
        return this;
	}

	setModifiedCount(modifiedCount: number) {
		this.modified_count = modifiedCount;
        return this;
	}

	setUpsertedId(upsertedId: string) {
		this.upserted_id = upsertedId;
        return this;
	}

	isAcknowledged(): boolean {
		return this.acknowledged;
	}

	getMatchedCount(): number {
		return this.matched_count;
	}

	getModifiedCount(): number {
		return this.modified_count;
	}

	getUpsertedId(): string {
		return this.upserted_id;
	}
}
